src/Features/Core/Portable/GenerateConstructorFromMembers/AbstractGenerateConstructorFromMembersCodeRefactoringProvider.ConstructorDelegatingCodeAction.cs

```diff
@@ -56,19 +56,25 @@ protected override async Task<Document> GetChangedDocumentAsync(CancellationToke
                 var codeGenerationService = languageServices.GetRequiredService<ICodeGenerationService>();
 
                 Contract.ThrowIfNull(_state.DelegatedConstructor);
-                var thisConstructorArguments = factory.CreateArguments(
-                    _state.Parameters.Take(_state.DelegatedConstructor.Parameters.Length).ToImmutableArray());
+                var thisConstructor = _state.DelegatedConstructor.ContainingType == _state.ContainingType;
+                var delegatedParameterNames = _state.DelegatedConstructor.Parameters.Select(p => p.Name).ToImmutableHashSet();
```

Comment:
this set seems to jsut be used for its names.  it doesn't really seem necessary.  the below line coudl just be:

`factory.CreateArguments(_state.Parameters.WhereAsArray(p => _state.DelegatedConstructor.Parameters.Contains(p)));`

right?

or, if that wouldn't work, can you explain why?  thanks!

Reply:
That wouldn't work because `_state.DelegatedConstructor.Parameters` come from ` : base(name, age)`, whereas `_state.Parameters` come from `public Employee(string name, string age, string salary, string role)` (example from the [test case](https://github.com/dotnet/roslyn/pull/63942/files#diff-eaeabcc51d3bc4c1219b8281400b921af35c9df7181857150f0524a69a8b99ccR1872)). Hence, they are completely different instances, but have the same name.

But resolved differently: 6664ea1ba806f528594251e87b2539a6214358cd